src/components/views/Instance/Edit/InstancePermissions.tsx

```diff
@@ -0,0 +1,862 @@
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React from "react";
+import { Alert, Col, Form, InputGroup, Tab, Tabs } from "react-bootstrap";
+import Button from "react-bootstrap/Button";
+import OverlayTrigger from "react-bootstrap/OverlayTrigger";
+import Tooltip from "react-bootstrap/Tooltip";
+import { FormattedMessage } from "react-intl";
+import { RouteComponentProps, withRouter } from "react-router-dom";
+
+import {
+    AdministrationRights,
+    ByondRights,
+    ChatBotRights,
+    ConfigurationRights,
+    DreamDaemonRights,
+    DreamMakerRights,
+    InstanceManagerRights,
+    InstancePermissionSetResponse,
+    InstancePermissionSetRights,
+    InstanceResponse,
+    RepositoryRights,
+    UserGroup,
+    UserResponse
+} from "../../../../ApiClient/generatedcode/generated";
+import InstanceClient from "../../../../ApiClient/InstanceClient";
+import InstancePermissionSetClient from "../../../../ApiClient/InstancePermissionSetClient";
+import InternalError, { ErrorCode } from "../../../../ApiClient/models/InternalComms/InternalError";
+import { StatusCode } from "../../../../ApiClient/models/InternalComms/InternalStatus";
+import UserClient from "../../../../ApiClient/UserClient";
+import UserGroupClient from "../../../../ApiClient/UserGroupClient";
+import { InstanceEditContext } from "../../../../contexts/InstanceEditContext";
+import {
+    hasAdminRight,
+    hasInstanceManagerRight,
+    hasInstancePermRight,
+    resolvePermissionSet
+} from "../../../../utils/misc";
+import { AppRoutes, RouteData } from "../../../../utils/routes";
+import ErrorAlert from "../../../utils/ErrorAlert";
+import InputField, { AnyEnum, FieldType } from "../../../utils/InputField";
+import { DebugJsonViewer } from "../../../utils/JsonViewer";
+import Loading from "../../../utils/Loading";
+
+interface IProps extends RouteComponentProps {}
+
+interface Permission {
+    readonly bitflag: number;
+    readonly currentVal: boolean;
+}
+
+interface IState {
+    errors: Array<InternalError<ErrorCode> | undefined>;
+    loading: boolean;
+    loadingPerms: boolean;
+    instanceNeedsReload: boolean;
+    tab: string;
+    users?: UserResponse[] | null;
+    groups?: UserGroup[] | null;
+    instancePermissionSets?: InstancePermissionSetResponse[];
+    selectedPermissionSetId: number;
+    userPermissions: InstancePermissionSetResponse;
+    currentPermissions?: InstancePermissionSetResponse | null;
```

Comment:
I tried figuring it out but I can't, what in the world is the difference between both of theses? Pretty much impossible to finish the review without knowing